src/admin/components/forms/field-types/getFilterOptionsQuery.ts

```diff
@@ -6,8 +6,8 @@ export const getFilterOptionsQuery = (filterOptions: FilterOptions, options: Omi
   const relations = Array.isArray(relationTo) ? relationTo : [relationTo];
   const query = {};
   if (typeof filterOptions !== 'undefined') {
-    relations.forEach((relation) => {
-      query[relation] = typeof filterOptions === 'function' ? filterOptions({ ...options, relationTo: relation }) : filterOptions;
+    relations.forEach(async (relation) => {
```

Comment:
Async await doesn't work properly inside forEach as far as I remember.

Reply:
I recommend using a regular for ( const x of y ) instead of forEach to avoid this JS bug.

Reply:
Changed the `.forEach` to be a `.map` using `Promise.all` I believe this will fix the bug.